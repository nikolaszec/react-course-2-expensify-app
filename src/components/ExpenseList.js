import React from 'react'
import {connect} from 'react-redux'
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'


export const ExpenseList = (props)=>(

    <div>
        {props.expenses.length===0?(
            <p>No expenses to display.</p>
        ):(
            props.expenses.map((expense)=>
            {return <ExpenseListItem key={expense.id} {...expense}/>}
            )
        )}
        
    </div>
)

const MapToProps = (state) => {
    return {
        expenses:selectExpenses(state.expenses,state.filters)
        
    }
}

export default connect(MapToProps)(ExpenseList)