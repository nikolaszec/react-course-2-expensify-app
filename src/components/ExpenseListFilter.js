import React from 'react'
import {connect} from 'react-redux'
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters';
import 'react-dates/initialize'
import {DateRangePicker} from 'react-dates'


export class ExpenseListFilter extends React.Component {

    state = {
        calendarFocus:null
    }

    onDatesChange = ({startDate,endDate})=>{
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }

    onFocusChange = (calendarFocus)=>{
        this.setState(()=>({
            calendarFocus
        }))
    }

    onTextChange = (e)=>{
        this.props.setTextFilter(e.target.value)
        }

        sortByChange = (e)=>{

            
            switch(e.target.value){
                case 'amount':
                this.props.sortByAmount()
                break;
 
                case 'date':
                this.props.sortByDate()
                break;
            }
         }

    render(){
     return (<div className="content-container">
      <div className="input-group">
      <div className="input-group__item">
      <input type= 'text' className="text-input"
        placeholder="Search expenses"
         value={this.props.filters.text} 
         onChange={this.onTextChange}/>
      </div>
      <div className="input-group__item">
      <select 
      className="select"
      value = {this.props.filters.sortBy}
  
      onChange={this.sortByChange}>
          <option value='date'>Date</option>
          <option value='amount'>Amount</option>
      </select>
      </div>
      <div className="input-group__item">
      
      <DateRangePicker 
      startDateId = 'start'
      endDateId ='end'
      startDate = {this.props.filters.startDate}
      endDate = {this.props.filters.endDate}
      onDatesChange= {this.onDatesChange}
      focusedInput = {this.state.calendarFocus}
      onFocusChange = {this.onFocusChange}
      numberOfMonths = {1}
      isOutsideRange = {()=>false}
      showClearDates = {true}
        />
      </div>
    </div>
    </div>)
    }
}



const mapStateToProps = (state) => {
    return {
        filters:state.filters
    }
}

const mapDispatchToProps = (dispatch)=> ({
setStartDate: (startDate)=> dispatch(setStartDate(startDate)),
setEndDate:(endDate) => dispatch(setEndDate(endDate)),
setTextFilter:(value)=>dispatch(setTextFilter(value)),
sortByAmount:()=>dispatch(sortByAmount()),
sortByDate:()=>dispatch(sortByDate())
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilter)