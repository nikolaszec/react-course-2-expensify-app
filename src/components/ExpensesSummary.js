import React from 'react'
import getTotalExpense from '../selectors/expensesTotal'
import {connect} from 'react-redux'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'
import {Link} from 'react-router-dom'


export const ExpensesSummary = ({expenseCount, expensesTotal})=>{

    const expenseWord = expenseCount===1?'expense':'expenses'
   const expensesTotalFormated  = numeral(expensesTotal).format('$0,0.00')

    return (
        <div className="page-header">
        <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{expensesTotalFormated}</span></h1>
        <div className="page-header__actions">
            <Link className="button-partial" to="/create">Add Expense</Link>
        </div>
        </div>
      </div>
    )

}

const mapStateToProps = (state)=> {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)

    return {
        expenseCount:visibleExpenses.length,
        expensesTotal:getTotalExpense(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)