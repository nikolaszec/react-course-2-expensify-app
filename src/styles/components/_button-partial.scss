.button-partial{
    font-size: $font-size-large;
    background-color:fade-out($color:$blue, $amount: .20);
    color: whitesmoke;
    padding: $s-size;
    display: inline-block;
    text-decoration: none;
    line-height: 1;
    font-weight: 300;
    border:none;
}

.button-partial--link {
    background: none;
    
}

.button-partial--remove{
    @extend .button-partial;
    background: #888888;
}