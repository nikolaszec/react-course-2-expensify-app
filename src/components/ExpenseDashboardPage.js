import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilter';
import ExpensesSummary from './ExpensesSummary';


const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseListFilter/>
    <ExpenseList ></ExpenseList>
  </div>
);



export default ExpenseDashboardPage;
