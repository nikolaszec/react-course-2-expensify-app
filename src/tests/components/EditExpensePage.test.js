import React from 'react'
import {shallow} from 'enzyme'
import {EditExpensePage} from '../../components/EditExpensePage'
import expenses from '../fixtures/expenses'

let editExpense, history, wrapper, removeExpense

beforeEach(()=>{
    editExpense = jest.fn();
    history = {push:jest.fn()}
    removeExpense = jest.fn()
     wrapper = shallow(<EditExpensePage 
        editExpense={editExpense}
         history={history} 
         removeExpense={removeExpense}
          expense={expenses[1]}
          />)
})

test('Should render edit page correctly', ()=>{
    expect(wrapper).toMatchSnapshot()
})



test('Should render edit page correctly with passed data', ()=>{
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1])
    expect(history.push).toHaveBeenLastCalledWith('/')
   // wrapper.setProps({expense: {id:expenses[2].id}});
expect(editExpense).toHaveBeenLastCalledWith(expenses[1].id, expenses[1])
   expect(wrapper).toMatchSnapshot()
})


test('Should remove expense from edit page correctly with button click', ()=>{
    wrapper.find('button').simulate('click')
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(removeExpense).toHaveBeenLastCalledWith({
        id:expenses[1].id
    })
  
})