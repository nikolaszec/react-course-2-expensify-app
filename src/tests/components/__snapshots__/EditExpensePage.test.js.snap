// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render edit page correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Edit Expense
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <ExpenseForm
      expense={
        Object {
          "amount": 200,
          "createdAt": 345600000,
          "description": "def",
          "id": "2",
          "note": "",
        }
      }
      onSubmit={[Function]}
    />
    <button
      className="button-partial--remove"
      onClick={[Function]}
    >
      Remove expense
    </button>
  </div>
</div>
`;

exports[`Should render edit page correctly with passed data 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Edit Expense
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <ExpenseForm
      expense={
        Object {
          "amount": 200,
          "createdAt": 345600000,
          "description": "def",
          "id": "2",
          "note": "",
        }
      }
      onSubmit={[Function]}
    />
    <button
      className="button-partial--remove"
      onClick={[Function]}
    >
      Remove expense
    </button>
  </div>
</div>
`;
